src/components/decorative-boxes/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react';
+import style from './style.module.css';
+type BoxProperties = {
+  color: `--color-${DefaultColors}${ColorModifiers}` | HexColorAsString;
+  position: `${VerticalPosition}-${HorizontalPosition}` | NumericPosition;
+  text?: string | undefined;
+};
+
+type NumericPosition = {
+  horizontalPosition: number;
+  verticalPosition: number;
+};
+type BoxSize = 'small' | 'medium' | 'large' | number;
+type VerticalPosition = 'top' | 'topish' | 'middle' | 'bottomish' | 'bottom';
+type HorizontalPosition = 'left' | 'leftish' | 'middle' | 'rightish' | 'right';
+type DefaultColors = 'primary' | `secondary${1 | 2 | 3 | 4}`;
+type ColorModifiers = '' | `__${'tint' | 'shade'}${1 | 2 | 3 | 4}`;
+type HexColorAsString = `#${string}`;
+
+export interface DecorativeBoxesProps
+  extends React.ComponentPropsWithoutRef<any> {
+  children: JSX.Element;
+  boxSize?: BoxSize;
+  box1Properties: BoxProperties;
+  box2Properties: BoxProperties;
+}
+
+const DecorativeBoxes: React.FC<DecorativeBoxesProps> = ({
+  children,
+  boxSize = 'medium',
+  box1Properties,
+  box2Properties,
+}) => {
+  const box1HorizontalDistance = useRef(
+    calculateDistanceToSide(
+      getNumericalBoxSize(boxSize),
+      box1Properties.position,
+      'left',
+    ),
+  );
+  const box1VerticalDistance = useRef(
+    calculateDistanceToSide(
+      getNumericalBoxSize(boxSize),
+      box1Properties.position,
+      'top',
+    ),
+  );
+  const box2HorizontalDistance = useRef(
+    calculateDistanceToSide(
+      getNumericalBoxSize(boxSize),
+      box2Properties.position,
+      'left',
+    ),
+  );
+  const box2VerticalDistance = useRef(
+    calculateDistanceToSide(
+      getNumericalBoxSize(boxSize),
+      box2Properties.position,
+      'top',
+    ),
+  );
+
+  return (
+    <div className={style['main-container']}>
+      <div
+        className={style['decorative-box']}
+        style={{
+          width: getNumericalBoxSize(boxSize) + '%',
+          backgroundColor: `var(${box1Properties.color})`,
+          left: box1HorizontalDistance.current + '%',
+          top: box1VerticalDistance.current + '%',
+          transform: `translate(
+            -${box1HorizontalDistance.current}%, 
+            -${box1VerticalDistance.current}%
+            )`,
+        }}
+      >
+        {box1Properties.text ? box1Properties.text : ''}
+      </div>
+      <div
+        className={style['decorative-box']}
+        style={{
+          width: getNumericalBoxSize(boxSize) + '%',
+          backgroundColor: `var(${box2Properties.color})`,
+          left: box2HorizontalDistance.current + '%',
+          top: box2VerticalDistance.current + '%',
+          transform: `translate(
+            -${box2HorizontalDistance.current}%, 
+            -${box2VerticalDistance.current}%
+            )`,
+        }}
+      >
+        {box2Properties.text ? box2Properties.text : ''}
```

Comment:
Dette kunne også vært enklere uttrykt slik

```suggestion
        {box2Properties.text ?? ''}
```